scripts/microbe_stage/microbe_editor/microbe_editor.as

```diff
@@ -617,18 +617,36 @@ class MicrobeEditor{
         if(action.cost != 0)
             if(!takeMutationPoints(action.cost))
                 return;
-        // We resize always and insert at the index so that
-        // if we undo something then add something, the new
-        // action is in the right spot on the array.
-        // since we only enqueue when we add new actions
-        actionHistory.resize(actionHistory.length()+1);
+
+        // Every time we make a new action if is in the middle of history of actions
+        // We have to be sure that this action is not a different from what we expect
+        // In the history, if it's different we have to erase all redo history
+        if( actionHistory.length() > 0 && (actionIndex < actionHistory.length() ) ) {
+
+            int q = int(action.data["q"]);
+            int r = int(action.data["r"]);
+
+            int nextQ = int(actionHistory[actionIndex].data["q"]);
+            int nextR = int(actionHistory[actionIndex].data["r"]);
+
+            int i =  actionIndex;
+
+            if( (q != nextQ) || (r != nextR)) {
```

Comment:
I'm failing to undestand why this is needed? Wouldn't logic like this work: if actionIndex is not at the end of action History, chop off items from actionHistory until it is at the end?

Reply:
You right, that check is useless, i commit the changes soon.